### PR TITLE
Add disableEventPropagation prop to prevent external side-effects

### PR DESCRIPTION
**Pull request by troycroz**
Currently some click events in react-datetime propagate up the DOM, and get picked up by Semantic's modal, causing the modal to close.

This gives the option to isolate react-datetime events, and not have them propagate externally.

I will open a PR to the source repo, but we can use our forked copy until the time when they accept our PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,6 +69,7 @@ render: function() {
 | **closeOnTab** | `boolean` | `true` | When `true` and the input is focused, pressing the `tab` key will close the datepicker.
 | **timeConstraints** | `object` | `null` | Add some constraints to the timepicker. It accepts an `object` with the format `{ hours: { min: 9, max: 15, step: 2 }}`, this example means the hours can't be lower than `9` and higher than `15`, and it will change adding or subtracting `2` hours everytime the buttons are clicked. The constraints can be added to the `hours`, `minutes`, `seconds` and `milliseconds`.
 | **disableOnClickOutside** | `boolean` | `false` | When `true`, keep the datepicker open when click event is triggered outside of component. When `false`, close it.
+| **eventBoundary** | `boolean` | `false` | When `true`, all synethetic click/touch/key events will stop native event propagation at the bounds of this component.
```
Comment (alexgs) [inline README.md]: Typo in "synethetic"

**Review by alexgs — APPROVED** [review]

**Review by AfroDevGirl — APPROVED** [review]

File: DateTime.js [hunk 2]
```diff
@@ -15,6 +15,12 @@ var viewModes = Object.freeze({
 	TIME: 'time',
 });
 
+var eventBoundaryEventTypes = Object.freeze([
```
Comment (scottrblock) [inline DateTime.js]: Maybe a future enhancement, but I could see people wanting to configure which of these this applies to if enabled?

**Review by scottrblock — APPROVED** [review]

File: DateTime.js [hunk 3]
```diff
@@ -416,6 +423,17 @@ var Datetime = createClass({
 		return props;
 	},
 
+	eventBoundaryProps: function () {
```
Comment (troycroz) [inline DateTime.js]: just realized that this whole method is static and should probably just be defined outside the component, where I defined the `eventBoundaryEventTypeHandlers`.

Thoughts?

Comment (scottrblock) [reply DateTime.js]: Probably a good idea. If one place makes it easier to spec, put it in that place

**Review by Brantron — APPROVED** [review]

**Review by AfroDevGirl — APPROVED** [review]
Looks great. Can't think of anything at the moment but maybe changing the frontend codebase to this fork is a good forcing function to redo the specs for our wrapper component.

**Pull request merged**